src/Features/CSharp/Portable/ExtractInterface/CSharpExtractInterfaceService.cs

```diff
@@ -85,10 +95,16 @@ internal override async Task<SyntaxNode> GetTypeDeclarationAsync(Document docume
                 }
             }
 
-            var updatedTypeDeclaration = typeDeclaration.WithBaseList(updatedBaseList.WithAdditionalAnnotations(Formatter.Annotation));
-            var updatedRoot = root.ReplaceNode(root.GetAnnotatedNodes<TypeDeclarationSyntax>(typeNodeAnnotation).Single(), updatedTypeDeclaration);
-            var solutionWithOriginalTypeUpdated = solutionWithFormattedInterfaceDocument.WithDocumentSyntaxRoot(docId, updatedRoot, PreservationMode.PreserveIdentity);
-            return solutionWithOriginalTypeUpdated;
+            return typeDeclaration.WithBaseList(updatedBaseList.WithAdditionalAnnotations(Formatter.Annotation));
```

Comment:
all this code seems pretty unnecessary.  We already have SyntaxEditor.AddBaseType that shoudl be able to do everything here.  It's unclear to me why we're reinventing all the code to do that.